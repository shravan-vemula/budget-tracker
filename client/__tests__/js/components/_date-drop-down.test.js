import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import DateDropDown from "js/_components/organisms/_date-dropdown";
afterEach(cleanup);

const setSelectedDate = jest.fn();
describe("DateDropDown Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<DateDropDown setSelectedDate={setSelectedDate} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should display menu after clicking", () => {
    const { getByRole, getByText } = render(
      <DateDropDown setSelectedDate={setSelectedDate} />
    );
    expect(getByRole("button")).toBe;
    fireEvent.click(getByRole("button"));
    expect(getByText("June,2020")).toBe;
    fireEvent.click(getByText("June,2020"));
  });
});
