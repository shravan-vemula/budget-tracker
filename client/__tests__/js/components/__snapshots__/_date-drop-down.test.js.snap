// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateDropDown Component renders correctly 1`] = `ShallowWrapper {}`;
